lib/gui/lvgl/lvgl_display.c

```diff
@@ -25,6 +25,8 @@ int set_lvgl_rendering_cb(lv_disp_drv_t *disp_drv)
 #endif
 		break;
 	case PIXEL_FORMAT_RGB_888:
+	case PIXEL_FORMAT_RGB_666:
+	case PIXEL_FORMAT_BGR_666:
```

Comment:
ILI9340 does not support RGB888. It only supports RGB565 and RGB666, even though it treats RGB666 as RGB888 when transmitting the data. What is your suggestion here?

Reply:
Yes the controller only supports RGB666 but the SPI transfer format is RGB888 (the lower 2 bits of each color are discarded) if you would only send 18-bits instead of 24-bits it would fail.As the display API is an abstract interface I would use RGB888 as this is the actual format that is transferred to the controller.

This code would even fail if a display controller comes along that really supports RGB666 on its bus and what do we need to do in that case break LVGL support for the ILI controller or not support the new controller?

Reply:
Changed back to RGB888. I added a note in the DT binding file about RGB888.